tensorflow/compiler/mlir/tensorflow/transforms/tensor_list_ops_decomposition.cc

```diff
@@ -574,9 +864,38 @@ LogicalResult HandleEmptyTensorListOp(
       return list.emitOpError("unknown tensor list element shape");
     }
   }
-  if (failed(cutil::CreateInitBufferValue(
-          element_shape, list.max_num_elements(), list, list.element_dtype(),
-          builder, &buffer))) {
+  Value list_size = list.max_num_elements();
+  auto list_size_op = list_size.getDefiningOp<TF::ConstOp>();
+  if (!list_size_op) return list.emitOpError("unknown max element count");
+  ElementsAttr listSizeAttr = list_size_op.value();
+  auto moduleOp = static_cast<Operation*>(list)->getParentOfType<ModuleOp>();
```

Comment:
There is implicit cast so you should be able to remove explicit cast here and just rely on the cast to happen at the callsite.